kernel/threads.c

```diff
@@ -244,6 +245,28 @@ int fork (void) {
   return new_tcb->thread_id;
 }
 
+void __attribute__((noreturn)) fork_daemon (void) {
```

Comment:
This shares a lot of code with fork() - make a fork_internal() that does everything but the scheduling and share it?
This applies to the spawn_new_thread() variant as well but there's less code there so meh.

Reply:
Done.